### ISSUE TITLE
CfC: Do not merge WebWorkers into HTML

### ISSUE THREAD

**Issue by chaals** [w3c/html#1468]
This is a CfC on the proposal

> The Working Group should not merge Web Workers into the HTML specification. Note that this would overturn the decision made following the CfC #1090 

Further discussion led to the proposal to revisit the decision. The main reasons given were
1. The expectation that merging it would improve the work rate looks unlikely to be true
1. Web Workers is not really linked to HTML and is not an HTML-only feature

Please respond to this CfC by adding a thumbs up, or a comment explaining (or adding a thumbs down to a comment explaining) why you disagree with the proposal, on or before 29 June 2018.

**Comment by awwright** [comment]
I work on some other Web technologies that don't touch HTML, like Semantic Web and JSON Schema. Consolidation, like this case, I believe has made it more difficult to integrate together alternative Web technologies like the ones I work on.

HTML is certainly the flagship hypermedia technology, but it is not the only one with all the necessary features. And there's many reasons not to use HTML: HTML lacks several important features, for example, forms still only support POST, and only allow submission of string key/string value pairs. Other new hypermedia technologies, like JSON Hyper-schema, are designed to allow a spectrum types like arrays or numbers, along with the full semantics that HTTP offers, like the PUT or PATCH methods, and many new headers.

Yet it's quickly becoming more and more difficult to avoid referencing HTML, when not otherwise necessary, because generic hypermedia features like WebWorkers or history.pushState are only accessible through HTML.

For a while, the HTML WG (formally or informally) followed the [Priority of Constituencies](https://www.w3.org/TR/2007/WD-html-design-principles-20071126/#priority-of-constituencies), where implementation ease generally takes priority over specification authoring. That is to say, it's the job of the specification authors to front-load work on behalf of implementors.

The exact implications of this are not well defined, however I would like to suggest this means breaking up specifications into logical parts, when it is possible to independently implement just that part (as is the case with WebWorkers), then if necessary, non-normatively explaining how the parts work as a whole.

The [architecture of the Web](https://www.w3.org/TR/webarch/) places HTML as a [media type](https://tools.ietf.org/html/rfc6838), which has specific connotations. The specification should define a concrete syntax, how to parse the syntax, how to handle errors, define a data model (the kind of data that can be parsed from the document), security considerations, and possibly how to visually render the data. HTML, for its part, does all of this.

However HTML is not an operating system. User agents are not always browsers, they do not always support scripting, sometimes they're even rendered by non-visual methods, or not at all.

Scripting APIs that have more to do with WebIDL-describable programming languages should be released as separate specifications.

In short, HTML is not and should not be The Web Browser Specification. It is a media type, and should be written as such to maximize adoption by the diverse spectrum of user agents that are out there.

**Comment by chaals** [comment]
This CfC passes. (Sorry for not closing it off sooner)
